pkg/envtest/komega/transforms.go

```diff
@@ -25,20 +25,20 @@ import (
 	gtypes "github.com/onsi/gomega/types"
 )
 
-// WithField gets the value of the named field from the object.
+// HaveField gets the value of the named field from the object.
```

Comment:
This was originally `WithField` because of the naming of the `WithTransform` in gomega, so this then matches that naming scheme. I don't have a strong opinion on which it should be called though, just wanted to open up for discussion in case anyone had some opinions about keeping this in line with gomega

Reply:
I changed it because it's a little nicer to read imo.
```
Expect(...).To(HaveField(..., Equal(...))
```
vs
```
Expect(...).To(WithField (..., Equal(...))
```

Reply:
Turns out this already exists: https://pkg.go.dev/github.com/onsi/gomega@v1.17.0#HaveField
I'll remove it.

Reply:
Oh sweet, they literally just introduced that!

Reply:
Nice catch!

Reply:
Unfortunately that one doesn't support pointers, which are used within the API structs quite a bit. With the new `Object()` function it's then quite important, so I'll leave it in for now. If we decide on using this API we can open an Issue/PR to Gomega and see if we can get pointer support.

> Nice catch!

Heh, I tried to `import . "github.com/onsi/gomega"` and it complained that `HaveField` already exists :smile:

Reply:
It would be good if we can improve the gomega version to support pointers, otherwise I'd suggest renaming this to something that doesn't conflict with the gomega function

Reply:
I created an issue on Gomega to suggest adding pointer support and will work on fixing there if we can get agreement from the maintainer https://github.com/onsi/gomega/issues/494

Reply:
I was about to do the same, thanks!